sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -301,12 +301,24 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
   def start(): StreamingQuery = startInternal(None)
 
   /**
+   * :: Experimental ::
+   *
    * Starts the execution of the streaming query, which will continually output results to the given
    * table as new data arrives. A new table will be created if the table not exists. The returned
    * [[StreamingQuery]] object can be used to interact with the stream.
    *
+   * Note, if the table does not exist, both V1 and V2 tables will respect the partitioningColumns
```

Comment:
We could consider using `@note` too but I think this is fine as is too ... as it's a bit long ..

Reply:
Follow the comment https://github.com/apache/spark/pull/30885#discussion_r547420351, the long paragraph separates into 2 parts.